Added Dark Mode Feature

MuhammadHasham23 opened the pull request:
I have added dark mode feature with a radio button on header.

The dark mode can be enabled and disabled as per user's choice. I've made a `style` object which changes the `background: black` and `color: white`. If `isDark` is set to `true`. Else the style won't be applied.

Since the header styles do not obey the styles injected on the parent `div`. We needed to pass the prop ( a function to trigger the change) down to `ToggleSwitch`. Which is the grand child of the parent component. So instead of passing down props through `Header`. Context API has been implemented with a `Provider` and `Consumer`. 

As it can be seen below,

![image](https://user-images.githubusercontent.com/17927649/79950400-7ddcc980-8490-11ea-9bcc-043970cf15db.png)


A new directory for contexts is made to host contexts in future as well, currently it hosts `styleContext`

naveen521kk commented:
Is there a test deploy?

MohammadHasham commented:
> Is there a test deploy?

Nope, just making sure that Codacy passes all tests for this PR. Once done it will be ready to be reviewed and merged.

saadpasta commented:
Thanks for the PR. Just some changes. 

1- Please add a hover effect on header buttons

2- Please add some styling to radio button to make sure it looks like a dark mode like some 
     moon emoji or take some inspiration from here **https://bchiang7.github.io/**

3- Style doesn't apply on **Achievements** and **Contact** Section heading

4- Remove the white border at the end of the page.

5- use light black theme just like here https://bchiang7.github.io/

![screencapture-localhost-3000-2020-04-22-12_29_11](https://user-images.githubusercontent.com/23307811/79953253-01001e80-8495-11ea-8ba9-37fbcb5b822c.png)

MohammadHasham commented:
Thanks for the review @saadpasta.

I'll fix the issues highlighted and get back to you.

kartik918 commented:
Could you add new screenshots @MohammadHasham?

Instead of complete black we could use a subtle grey like ``#181a1b`` or ``#1c1d1f`` . If you could send a screenshot with one of these too, that would be great!

And about the contacts section graphic, we could move it to SVG so it sits right with the rest of the illustrations. You can find it by typing "envelope in the search bar [here](https://undraw.co/search). (The link isn't shareable directly)

This is a great addition to the project @MohammadHasham , thanks for providing it!

naveen521kk commented:
@MohammadHasham I found that there is a option for Twitter dark mode. Please check it out [here](https://saurabhnemade.github.io/react-twitter-embed/?path=/story/twitter-timeline-embed--timeline-dark-theme). You could change the that when the button toggles. 😀

kartik918 commented:
As @naveen521kk said, for an effective dark mode we would need one CSS file that overwrites all relevant CSS instead of just the background. This could be switched in and out on button press.

You also need to update all individual elements to avoid design inconsistencies like the twitter section or the blogs one.

You could use ``!important`` on properties that aren't overridden on the new CSS change. Hope this helps you write a more efficient solution.

MohammadHasham commented:
Thanks a lot for the feedback. I'll improve the solution and work on the inconsistencies @naveen521kk @kartik918

MohammadHasham commented:
The following are the highlights of last code shipped to PR:

- Added purple color on hover for navigation bar.
- Instead of just changing the background color now there are separate classes in stylesheets that gets activated when the button is toggled. 
- Twitter dark modes is supported as well.
- Minor design fixes for the toggle button.
- Other minor fixes for colors


 Here is an image attached.

![screencapture-localhost-3000-2020-05-01-02_30_59](https://user-images.githubusercontent.com/17927649/80796732-681a8300-8bb9-11ea-9f9c-7d01d6b0261d.png)

naveen521kk commented:
@MohammadHasham Add `transparent={true} noScrollbar={true}` to the twitter component. This will remove the scrollbar and make the twitter component transparent. And then it would look good. Also, add the `className="main"` to the top-level div to make it would be aligned even with the other components.

MohammadHasham commented:
@naveen521kk a

> @MohammadHasham Add `transparent={true} noScrollbar={true}` to the twitter component. This will remove the scrollbar and make the twitter component transparent. And then it would look good. Also, add the `className="main"` to the top-level div to make it would be aligned even with the other components.

@naveen521kk adding `className="main"` to the top level div makes website centered. Am I doing something wrong?

It looks like this:

![screencapture-localhost-3000-2020-05-01-18_01_20](https://user-images.githubusercontent.com/17927649/80807141-d10ff400-8bd5-11ea-840c-88e692151723.png)

kartik918 commented:
Hey @MohammadHasham this looks amazing! There's still inconsistencies left with some components (like I see white blog cards shadow from your screenshot) but other than that this looks great.

MohammadHasham commented:
Sure @kartik918 can you let me know those, so I can fix them. By the way, thank for the review.

kartik918 commented:
If you could deploy your branch I could provide better help in debugging. All I could make out was the card shadow and as Naveen said, the Twitter component.

I would advise against changing twitter layout as #107 introduces a lot of changes that would need time to fit right with the code, one of the changes being layouts.

Transparent Twitter looks good but keep the header visible, i.e "Tweets by <username>".

MohammadHasham commented:
I've deployed branch from forked repository. It would be great if you guys can have a look and let me know what exactly can be fixed/improved.

Here is the link:
https://eager-golick-372f8f.netlify.app/

@naveen521kk @kartik918 

cc: @saadpasta

naveen521kk commented:
@MohammadHasham There was some error on mobile. Check the screenshot below.
![Screenshot_20200508-170941](https://user-images.githubusercontent.com/49693820/81402453-0d4fd100-914f-11ea-9cb9-63298e904688.png)

kartik918 commented:
I just took a quick test on my phone and it seems all the text is white, even in light mode.

The dark mode works fine and looks great!

There's also an issue with the menu on smaller screens, you might want to look into that too.

Lastly there's 3 tiny dots near the top left of the toggle button which shouldn't exist in the first place.

You've made good progress from before but there's still some more tweaking left!

naveen521kk commented:
Further, I think we can set a cookie to check whether the user previously had used dark mode. Next time when he loads the page we can load dark mode if he had previously enabled it. It would good. And I think the text is set to white color even after toggling to normal mode. So it must be fixed.

kartik918 commented:
I'm not for using cookies, mainly because of two reasons:

- using cookies means more permissions and notifying the user about them. If a portfolio uses cookies, no one would trust it, because there is no reason for one.

- portfolio is something that you visit once for a person, rarely ever twice for that same person. Saving a preference for that one tiny chance that they might seems impractical. 

What do you think @MohammadHasham  @saadpasta ?

MohammadHasham commented:
I think we can utilize local storage for that. We can keep a simple boolean `isDark: true | false`.

kartik918 commented:
We can , but should we?

MohammadHasham commented:
I've added the fixed for mobile devices. The menu works fine now, also added some other tweaks. I couldn't find out the three dots mentioned by @kartik918. I have implemented some other minor tweaks.

The dark mode preference has also been added. I have used 

`window.matchMedia('(prefers-color-scheme: dark)');`

The new deployment to the feature branch has been deployed: https://eager-golick-372f8f.netlify.app/

kartik918 commented:
``window.matchMedia()`` is working as it should and looks neat. I see some minor UI issues like the following:

Light Mode:
![image](https://user-images.githubusercontent.com/48270786/81494789-1cf32500-92c9-11ea-8a12-078d628a1487.png)

Dark Mode
![image](https://user-images.githubusercontent.com/48270786/81494797-28dee700-92c9-11ea-8897-b58a241882cf.png)

I think we can move the button in the nav bar so on smaller screens it doesn't take an anomalous amount of space on the landing.

naveen521kk commented:
I think it's better to a floating button on the left side of screen to toggle dark mode. That would look neat than placing on NavBar. What's your opinion @kartik918

kartik918 commented:
> 
> 
> I think it's better to a floating button on the left side of screen to toggle dark mode. That would look neat than placing on NavBar.

Floating left would collide with the ``<a class="logo">`` and page content (similar to how float to top does minimally). I don't see many options for placing the button in our current layout other than nav menu. 

If you have any other ideas to place it, let's discuss them.

naveen521kk commented:
I actually meant float left bottom parallel to back to top button. Or a better place would be in footer.

kartik918 commented:
Footer isn't ideal since once you've reached the end of a portfolio, the experience is over. 
Floating left parallel to Top button would take up too much footer space on the screen.

If we move the button inside the nav menu, it will look good for bigger screens and solve the space problem for other viewports too, but would be hidden away for smaller screens. 

We can make a radio button to save space maybe and move it left of the menu button? It wouldn't look good though.

naveen521kk commented:
On being in the top of page the top button won't show. Can we place the dark button over there in the top. That would do I believe.

kartik918 commented:
Let's just move it to nav menu, if we're setting the theme based on system pref, we don't need to explicitly show an option.

Thumbs up or down to vote I guess. @saadpasta, would appreciate an opinion on this.

saadpasta commented:
@kartik918 I guess you are right Nav is the right option

naveen521kk commented:
Can we do something like https://pilab.dev/ in our site? A bar in the bottom which hides automatically and place our button there?

MohammadHasham commented:
How shall we implement it? Do we have a final call on this. 

Let's get this done.

saadpasta commented:
@MohammadHasham Let's put it in the Navbar. 

@naveen521kk what you are suggesting is great I liked but it would be hard to implement it right now we can keep it for the redesign.

naveen521kk commented:
And @MohammadHasham Can you please take a screenshot on how GitHub repo card looks in dark mode? Its not visible in your test deploy.

MohammadHasham commented:
I have added the button in the navbar, but as @naveen521kk pointed out that the github projects were not being shown.

Upon further inspection, I came to know that GraphQL API is failing, here is the screenshot attached.

![image](https://user-images.githubusercontent.com/17927649/82578135-6be06a80-9ba5-11ea-806c-0ee39f7ec670.png)

@saadpasta can you look into it, I have added my token and username as well.

MohammadHasham commented:
The toggle button has been now moved to navbar. Also, background effect now is also being applied to open source projects. 

Hovering on open source projects make them purple.

Feel free to dabble around the deployment for these new changes:

https://eager-golick-372f8f.netlify.app/


Here is the image attached:
![screencapture-localhost-3000-2020-05-21-22_41_08](https://user-images.githubusercontent.com/17927649/82588701-9d146700-9bb4-11ea-8ab7-46bf46bdf1ee.png)


I haven't added my github token so the live deployment might not show the open source projects.

cc: @saadpasta @naveen521kk @kartik918

naveen521kk commented:
Check on hover on the buttons @MohammadHasham . It turn fully white in dark mode which *should* not happen. Make the text black on hover.

naveen521kk commented:
@kartik918 I like _white_ shadows than the _purple_  one. White shadows look better. 

And @MohammadHasham it's time for this PR to be merged.

kartik918 commented:
Okay I'll ping @saadpasta too so we can come to a consensus about going with white shadows.

Also, there's still quite some work to be done so we can't merge the PR in it's current state.

MohammadHasham commented:
Sure. I am on it, I'll get the changes up and ready! asap.

saadpasta commented:
@kartik918 I also think white shadow looks better

saadpasta commented:
@MohammadHasham Please resolve conflicts and get the latest changes from the master branch🙂

MohammadHasham commented:
I have merged `master` with `feat/dark-mode` and added breakpoints for toggle switch.

@saadpasta please have a review on it, and let me know if there is anything to be updated.

naveen521kk commented:
Why is the deployment failed? @MohammadHasham

saadpasta commented:
@MohammadHasham your branch is still 17 commits behind `developerFolio/master` branch

MohammadHasham commented:
> @MohammadHasham your branch is still 17 commits behind `developerFolio/master` branch

Done with merging from master to feature branch.

MohammadHasham commented:
How are we going to proceed further on this?

@naveen521kk @kartik918
cc: @saadpasta

naveen521kk commented:
Just fix changes requested by @kartik918 in this [comment](https://github.com/saadpasta/developerFolio/pull/92#pullrequestreview-417072398). Then we will be good to go.

kartik918 commented:
I updated my comment slightly to reflect on the stuff left to change @MohammadHasham.

MohammadHasham commented:
I have done the required fixes. But the contact image is on which I am skeptical about, the background in the PNG for dark illustration is still white. However, I have used that still.

@kartik918 can you see if that is fine.

kartik918 commented:
[This](https://github.com/saadpasta/developerFolio/files/4747869/contact.zip) is the SVG version of the contact graphic, replace it with this and it will help out in a full dark mode. 

Let me know if you need any more help, also remember to rebase before your next commit so we can see the latest changes.

saadpasta commented:
@kartik918 @MohammadHasham We can use Dark Mode Option Something like this in 
[React Blog Github](https://saadpasta.github.io/react-blog-github/#/) and remove it from the Header. What do you suggest? 

It would solve both problems clustering in Header and Styling Issues and It would be present when we scroll down also.

kartik918 commented:
Yep good idea @saadpasta and I think Naveen suggested this too. 

The only problem is we have a scroll to top button who's not gonna be happy if dark mode takes it place.

saadpasta commented:
@kartik918 how about we give the right-hand place to Go to top button and left-hand place to the dark mode

naveen521kk commented:
@saadpasta That could just make clumsy on small screens! 

Also, I like using the colours rather than emoji we can use colours as @kartik918 said. Also the below one looks good.
![switch](https://user-images.githubusercontent.com/48270786/84066726-fa494e80-a9e3-11ea-8ca7-0f3a601ef158.png)

kartik918 commented:
> I think it's better to a floating button on the left side of screen to toggle dark mode. 

Discussed this too @saadpasta haha. I'm open for new suggestions or solutions though. If nothing comes to mind, then I guess we have to go with the header one. 

Also I've recently opened an issue that will make the navbar sticky so it will appear on screen while scrolling too.

kartik918 commented:
@MohammadHasham just checking to see if you're still working on this. It's okay if you need more time or help.

MohammadHasham commented:
Yes @kartik918 . I am working on this, I'll update it today or max by tomorrow.

saadpasta commented:
@MohammadHasham Any Updates?

naveen521kk commented:
@MohammadHasham just checking to see if you're still working on this.

MohammadHasham commented:
I will try to complete it by the end of this weekend, was caught up in the middle of something. 

cc: @naveen521kk @saadpasta

saadpasta commented:
@MohammadHasham bro look's like this PR has taken lot of time and there and other PR depended on this I hope you complete this soon 
Thanks for your patience ❤️

saadpasta commented:
@kartikchou @naveen521kk What are the remaining issue's in this PR.

kartikchou commented:
Hey @saadpasta, are you taking this up? I think we should clear some of our backlog of PRs this week.

The last changes I requested were at [this comment](https://github.com/saadpasta/developerFolio/pull/92#pullrequestreview-426505242), no newer commits have been added since so this should be a good start to pickup from!

saadpasta commented:
@kartikchou I see there is draft PR I guess we should merge that first

kartikchou commented:
@saadpasta yep, I'm working on it. I'll open a PR superseding it by today.

saadpasta commented:
I got a message from @MohammadHasham he would compete this PR by tomorrow evening. He was caught up with job switching.

naveen521kk commented:
Doing the things  @kartikchou mentioned would do!

MuhammadHasham23 commented:
Apologies for the delay guys. I have made the fixes as follows:

- Removed the emoji from toggle switch.
- Replaced the contact form image with dark image.
- Fixed toggle button on header.

I was trying to rebase master after fetching the current master, but there are so many conflicts now, and what I assume is that doing a rebase would override some of the important changes that have been done on master, as the feature branch would supersede (Please correct me if I am wrong, that is just my understanding).

What should be the approach here @saadpasta @naveen521kk @kartikcho

naveen521kk commented:
The one I did with #107 is to pull all the commits from master and resolved merge conflicts manually. Idk if there is a better approach than this. I spent around 30 minutes on that. But I think this PR won't take that much time. 
What I tried was 
```sh
git pull https://github.com/saadpasta/developerFolio.git master
```
in the branch, I was working on. Finally, it would create Merge conflicts branch and finally, it would commit it. Or better you can use the method Github Tells you.

kartikcho commented:
As Naveen mentioned, you will have to resolve conflicts manually to make it sit right with the master but don't worry, try resolving conflicts through vscode as it shows a side by side comparison of incoming changes from master and what was there before. It won't take long probably.
Will take a look at the code when this is rebased and vercel opens a preview deployment.

MuhammadHasham23 commented:
Done with rebasing this branch with master.I see there are some checks failing for vercel, how can I know and fix them @kartikcho

kartikcho commented:
It seems to be an outdated deployment, I'm not sure how I can fire it again to work. @saadpasta ?

naveen521kk commented:
Also, looks like merging is not completed. I do see the font awesome here. Please do check it. 
When I did #107 this didn't happen weird.

saadpasta commented:
@MuhammadHasham23 I have deployed again on vertical and it's looking good. I guess you need to rebase again we still have Font Awesome.

saadpasta commented:
@naveen521kk can you help with rebasing this branch to master

naveen521kk commented:
@saadpasta I am very sorry I was unable to do it, I was very busy :(

saadpasta commented:
> @saadpasta I am very sorry I was unable to do it, I was very busy :(

No problem bro I understand

saadpasta commented:
@MuhammadHasham23 @naveen521kk @kartikcho 
Re Basing has been done to master branch have a look at Dark Mode. Review the PR see If there are any changes needed. There were lots of conflicts so I might have missed something.

MuhammadHasham23 commented:
What else do we have left on this @saadpasta ?

saadpasta commented:
@MuhammadHasham23 I guess we are done

saadpasta commented:
![123](https://user-images.githubusercontent.com/23307811/88707658-692f5380-d0c7-11ea-9c2d-c095b9acca9c.PNG)
Just need to fix the color of the subtitle of contact

saadpasta commented on src/containers/twitter-embed/twitter.js:
```diff
@@ -1,42 +1,36 @@
-                    borderColor= "#fff"
-                    noFooter= {true}
-                    onload= {timeOut()}
-                  />
-                  </div>
-          </div>
-        </Suspense>
-  );  
-	} else {  
+			<Fade bottom duration={1000} distance="20px">
+				<div class="tw-main-div" id="twitter">
+					<div className="centerContent">
+						<TwitterTimelineEmbed
+							sourceType="profile"
+							screenName={twitterDetails.userName}
+							options={{ height: 400, width: { widthScreen } }}
+							placeholder="Can't load? Check privacy protection settings"
+							autoHeight={false}
+							key={isDark ? "1" : "2"}
+							theme={isDark ? "dark" : "light"}
+							borderColor="#fff"
+							transparent={true}
+							noScrollbar={true}
+							noFooter={true}
+						/>
+					</div>
+				</div>
+			</Fade>
+		);
+	} else {
 		return null;
 	}
 }
```
We have removed suspense and functionality for twitter Iframe Loading. 
I'll have a look into this.

naveen521kk commented:
![image](https://user-images.githubusercontent.com/49693820/88796416-35cff000-d1bf-11ea-9445-92f9dbd30e5b.png)
This section is missing dark mode @MuhammadHasham23

naveen521kk commented:
The colour of the SVG of the map marker should be changed. while in dark mode.
![image](https://user-images.githubusercontent.com/49693820/88796501-64e66180-d1bf-11ea-8a1d-28e34e1504a2.png)

naveen521kk commented:
These shadows look pretty odd to me it would be better if there is no shadow.
![image](https://user-images.githubusercontent.com/49693820/88796669-a7a83980-d1bf-11ea-89aa-d436e3f1a432.png)

naveen521kk commented:
Changing the colour of `Saad Pasta` would be good. @kartikcho I need your suggestion on this.
![image](https://user-images.githubusercontent.com/49693820/88796724-c3134480-d1bf-11ea-92bf-e71e3700f5a1.png)

saadpasta commented:
> Changing the colour of `Saad Pasta` would be good. @kartikcho I need your suggestion on this.
> ![image](https://user-images.githubusercontent.com/49693820/88796724-c3134480-d1bf-11ea-92bf-e71e3700f5a1.png)

I think white looks good

MuhammadHasham23 commented:
Okay, I'll get this done on weekend.

naveen521kk commented on src/components/header/Header.js:
```diff
@@ -30,21 +35,22 @@ function Header() {
           </li>
           <li>
-            <a href="#blogs">Blogs</a>
+            <a href="#achievements">Achievements</a>
           </li>
           <li>
-            <a href="#talks">Talks</a>
+            <a href="#contact">Contact Me</a>
           </li>
           <li>
-            <a href="#contact">Contact Me</a>
```
**Question**: Why the order was changed? It has to in chronological order right? #116

saadpasta replied:
You are right I guess we should change the order to which the components are listed

naveen521kk commented on src/containers/Main.css:
```diff
@@ -0,0 +1,5 @@
+.dark-mode {
+  background-color: #171c28;
+  color: white;
+  transition: "0.1s";
+}
```
**Question**: Can't this fit into `index.css` or `app.css`?

saadpasta replied:
We can fit these into index and app CSS but let make sure we keep them as clean as possible and use the Main CSS as our css for all the Components.

naveen521kk commented on src/containers/contact/Contact.js:
```diff
@@ -1,29 +1,31 @@
                 </div>
                 <div className="contact-image-div">
-                    <img alt="Saad Working" src={require("../../assets/images/contactMail.webp")}></img>
+                    <img alt="Saad Working" src={require(`../../assets/images/${isDark ? 'contactMailDark.svg' : 'contactMail.webp'}`)}></img>
```
**Question:** Why not we use SVG for both dark and light mode?

saadpasta replied:
We have webp for light mode so we can you that and for dark mode, if we are using SVG because we don't have webp image

naveen521kk commented on src/containers/projects/Projects.js:
```diff
@@ -1,22 +1,23 @@
-
-
+import { openSource,socialMediaLinks  } from "../../portfolio";
+import { Fade } from "react-reveal";
```
**Nitpick**: Unnecessary Import.

naveen521kk commented on src/containers/skills/Skills.js:
```diff
@@ -1,26 +1,28 @@
-        <div className="skills-text-div">
-          <h1 className="skills-heading">{skillsSection.title} </h1>
-          <p className="subTitle skills-text-subtitle">{skillsSection.subTitle}</p>
+        <div className={"skills-text-div"}>
```
**Question**: Do we need curly braces here shouldn't it be
```suggestion
        <div className="skills-text-div">
```

saadpasta commented on src/components/githubRepoCard/GithubRepoCard.css:
```diff
@@ -69,4 +69,15 @@
+}
+.dark-card-mode:hover {
+  background-color: #55198b !important;
 }
```
Why a new line?

naveen521kk replied:
Always there should be a new line and the end of a file as in whole codebase.

kartikcho commented:
White `Saad Pasta` (haha) looks great I think! 
Sorry, I haven't been able to review recent activity due to busy weeks.

naveen521kk commented:
The twitter section looks a bit off
![image](https://user-images.githubusercontent.com/49693820/88807795-cb737b80-d1cf-11ea-9319-8b313859448a.png)
Using transparent twitter would be good.

MuhammadHasham23 commented:
I've added the relevant changes can you check @naveen521kk

naveen521kk commented:
@MuhammadHasham23 The transparent twitter component is not yet fixed.

naveen521kk commented:
A small suggestion, Can you change the colour of the highlighted one(The X) to white in Dark Mode?
![image](https://user-images.githubusercontent.com/49693820/89712785-c4f3b980-d9b0-11ea-8683-e17b7d690ae7.png)

naveen521kk commented:
**Issue: ** `Saad Pasta` name should not break in medium size viewport.
![image](https://user-images.githubusercontent.com/49693820/89712828-fd939300-d9b0-11ea-8a15-5f6de9ecd3ce.png)

naveen521kk commented:
**Suggestion** Change the colour of Repo Symbol to White.
![image](https://user-images.githubusercontent.com/49693820/89712944-c70a4800-d9b1-11ea-9fde-00f95b545a18.png)

**Suggestion** Change the colour of Map Symbol to White.
![image](https://user-images.githubusercontent.com/49693820/89713015-544d9c80-d9b2-11ea-9002-add1f016b745.png)


**Issue** This Colour is not changed in Profile Section.
![image](https://user-images.githubusercontent.com/49693820/89712908-87436080-d9b1-11ea-8041-a61af9f39b83.png)

naveen521kk commented:
**Suggestion** White shadow would be better for talk section.
![image](https://user-images.githubusercontent.com/49693820/89713070-d0e07b00-d9b2-11ea-9c51-adb4afe56569.png)

MuhammadHasham23 commented:
sure thing. I'll get that fixed @naveen521kk

kartikcho commented:
> 
> 
> **Issue**: `Saad Pasta` name should not break in medium size viewport.

This isn't an issue related to the dark mode and is more related to the header. Maybe we can open a separate issue for this @naveen521kk ?

naveen521kk commented:
@MuhammadHasham23 What's your progress on this?

MuhammadHasham23 commented:
> @MuhammadHasham23 What's your progress on this?

Possibly, by the end of the upcoming week.

MuhammadHasham23 commented:
@naveen521kk I have done the changes you prescribed above, can you check and let me know if that are fine?

kartikcho commented on src/components/talkCard/TalkCard.css:
```diff
@@ -1,57 +1,96 @@
+  box-shadow: 0 20px 40px rgba(0, 0, 0, 0.08);
+}
+.dark-rectangle {
+  box-shadow: 4px 1px 20px 0px #ffffff;
```
I don't think we need this, the shadow size isn't the same as light mode and if we do want to go with a smaller shadow, the shadow is off by a few pixels. 
`0px 0px 20px 0px` seems to fit right the box right with the following changes.

Before:

![image](https://user-images.githubusercontent.com/48270786/91814109-47d6ff80-ec51-11ea-82e0-33215b6a1145.png)


After:

![image](https://user-images.githubusercontent.com/48270786/91814074-3aba1080-ec51-11ea-8b74-f93a0931f783.png)

kartikcho commented:
Also, `blog-subtitle` and location svg doesn't switch to white. 

![image](https://user-images.githubusercontent.com/48270786/91814198-65a46480-ec51-11ea-84c3-ff932a9b3018.png)

But other than that everything else looks dope! Happy to see this so close to being complete :)

naveen521kk commented on src/containers/twitter-embed/twitter.css:
```diff
@@ -16,7 +16,12 @@
     font-size: 56px;
     font-weight: 100 !important;
 }
+
+.timeline-viewport-dark {
+  background-color: #171c28;
+}
```
Is this used anywhere?

naveen521kk commented on src/containers/greeting/Greeting.css:
```diff
@@ -11,12 +11,20 @@
   margin-top: 2rem;
 }
 
+.dark-mode {
+  color: white;
+}
+
```
It looks like repeated,

naveen521kk commented on src/containers/skills/Skills.css:
```diff
@@ -1,6 +1,11 @@
 .skills-main-div {
   display: flex;
 }
+
+.dark-mode {
+  color: white !important;
+}
+
```
It looks like repeated.

naveen521kk commented on src/components/blogCard/BlogCard.css:
```diff
@@ -1,9 +1,16 @@
+.dark-mode {
+  color: white !important;
+}
+
```
It looks like repeated.

naveen521kk commented on src/components/achievementCard/AchievementCard.css:
```diff
@@ -1,3 +1,8 @@
+
+.dark-mode {
+  background-color: #171c28;
+}
+
```
It looks like repeated.

MuhammadHasham23 commented:
I have removed redundant css as mentioned by @naveen521kk above. I couldn't find the code block related to `.blog-subtitle` in the code. If you can let me know where it is, that would be great.

cc: @naveen521kk @kartikcho

kartikcho commented:
@MuhammadHasham23 `.blog-subtitle` is in `src/containers/blogs/Blog.css` & the `.subTitle` class is in `App.css`.

 `.blog-subtitle` seems to be reused in the contact section, we should switch it to a distinct class of it's own to avoid confusion like this in the future.

MuhammadHasham23 commented:
@kartikcho I have added `.blog-subtitle` in css files for components, where it was being used.

MuhammadHasham23 commented:
is there anything left on this PR?

cc: @naveen521kk @kartikcho

kartikcho reviewed (approved):
My review is an old one so I haven't kept up with it but you've addressed all of the requested changes from @saadpasta and @naveen521kk so I don't have anything more to add. 

You should rebase with master, resolve conflicts and see if the dark mode sits right with the new section in #250. Making it adapt to the new changes shouldn't be a hassle. Thanks again for working on this and good work!

kartikcho commented on src/containers/greeting/Greeting.css:
```diff
@@ -11,12 +11,20 @@
   margin-top: 2rem;
 }
 
+/* .dark-mode {
```
Remove this comment if it isn't in use.

MuhammadHasham23 commented:
I've rebased with master and done changes to make sure the dark mode sits in right. Please let me know if there are any other changes yet to be made.

cc: @saadpasta @naveen521kk @kartikcho

kartikcho commented:
Sounds good! Could you update the preview deployment @saadpasta and check why it needs permission to deploy sometimes?

saadpasta commented:
![FireShot Capture 025 - Saad Pasta Portfolio - developer-folio-pdcczrcrn vercel app](https://user-images.githubusercontent.com/23307811/94896768-b523bd80-04a7-11eb-8d82-07b7f28e435f.png)
Fix GithubRepoCard looks like some things are duplicated

saadpasta commented:
> Sounds good! Could you update the preview deployment @saadpasta and check why it needs permission to deploy sometimes?

Looks like I need to review the pull request to deploy adding you guys to vercel so you could also fire up deployment

kartikcho commented:
Everything blends perfectly @MuhammadHasham23! Just remove the 'double render' like Saad said and we should be good. What do you think about removing the repo size information since that doesn't seem exactly useful?

naveen521kk commented:
One thing I found going was not matching was hover part in experience card.
Also, twitter card colour is not matched with the background,
Other than that this PR looks good to me.
![image](https://user-images.githubusercontent.com/49693820/94927692-4151d700-04e0-11eb-826e-80e9dca1c1c6.png)

kartikcho commented:
>  hover part in experience card

There is no hover effect for it, might be a decision but I'm okay with it. We can go for a subtle white shadow otherwise like other cards.

naveen521kk commented:
> Are you suggesting for it to be transparent or any other color? I think I can improve on this in a follow up PR, let's get this merged ASAP for now so we can get the other PRs in too!

It should be transparent. It's not actually stopping me though. Make a new issue and we are good to go.

naveen521kk commented:
Also, wait I will review the code at once.

MuhammadHasham23 commented:
I have fixed the multiple Github cards render. Can you have a review @saadpasta ?

cc: @naveen521kk @kartikcho

naveen521kk commented on .all-contributorsrc:
```diff
@@ -236,6 +236,44 @@
         "code"
       ]
+    },
+    {
+      "login": "vandana1499",
+      "name": "unbeat",
+      "avatar_url": "https://avatars2.githubusercontent.com/u/29394600?v=4",
+      "profile": "https://github.com/vandana1499",
+      "contributions": [
+        "doc"
+      ]
+    },
+    {
+      "login": "lARSHADl",
+      "name": "Arshad Ahmed",
+      "avatar_url": "https://avatars3.githubusercontent.com/u/45604332?v=4",
+      "profile": "https://github.com/lARSHADl",
+      "contributions": [
+        "doc"
+      ]
+    },
+    {
+      "login": "Ergouzii",
+      "name": "Xiaohui Liu",
+      "avatar_url": "https://avatars1.githubusercontent.com/u/33507446?v=4",
+      "profile": "http://xiaohuiliu.me",
+      "contributions": [
+        "doc",
+        "code",
+        "design"
+      ]
+    },
+    {
+      "login": "Seungyeon-Lee",
+      "name": "Seungyeon-Lee",
+      "avatar_url": "https://avatars1.githubusercontent.com/u/26589915?v=4",
+      "profile": "https://seungyeon-lee.github.io/",
+      "contributions": [
+        "code"
+      ]
```
Why is this file in this? If it a git related thing I suggest you to 
- make a new branch from master and check into it.
```sh
git checkout -b <new-branch-name> saadpasta/master
```
if saadpasta is not know to git than
```sh
git remote add saadpasta https://github.com/saadpasta/developerFolio.git
```
and then run previous command.

- Squash and merge this branch into that new branch
```sh
git merge --squash origin feat/dark-mode
```
- Commit it and force push to this branch.
```sh
git commit -m "Added Dark Mode Feature"
git push -f origin <new branch name>:feat/dark-mode
```

This would remove all of this issue. But it will remove all previous code review which isn't a worry for me. @kartikcho or @saadpasta do you have any objection in doing that?

kartikcho replied:
That _should_ fix all the issues but seems like an unconventional way of dealing with this.
 
Wouldn't merging into the new branch still show these changes? Because this branch is on par with master too so the new branch with all the changes from here would also be dealing with the same @naveen521kk.

naveen521kk replied:
> Wouldn't merging into the new branch still show these changes? Because this branch is on par with master too so the new branch with all the changes from here would also be dealing with the same 

I don't think so. Doing those steps would just mean that we have a single commit on this whole PR and online diff between master and this is shown. Pretty much it is "unconventional " but actually far easier than rebasing the whole branch with the master which may create some issue not rebased properly.

MuhammadHasham23 replied:
I have already rebased with master btw.

saadpasta replied:
@MuhammadHasham23 Looks like you have rebased the branch with the master branch on your forked repo you need to rebase master with developerFolio master. 
Try reading this 
https://medium.com/@topspinj/how-to-git-rebase-into-a-forked-repo-c9f05e821c8a

saadpasta replied:
@MuhammadHasham23 Let's complete this PR before the end of this month ✌✌

MuhammadHasham23 replied:
Yeah sure, I'll wrap it up tomorrow (Saturday) @saadpasta

MuhammadHasham23 replied:
@saadpasta I did rebased with the upstream branch (i.e; from the original repo).

naveen521kk commented on src/containers/projects/Projects.js:
```diff
@@ -1,18 +1,20 @@
 import { openSource, socialMediaLinks } from "../../portfolio";
-
-
+import { Fade } from "react-reveal";
```
Nitpick: Unnecessary Import.

MuhammadHasham23 replied:
Done

naveen521kk commented on src/containers/greeting/Greeting.css:
```diff
@@ -11,12 +11,20 @@
   margin-top: 2rem;
 }
 
+/* .dark-mode {
+  color: white;
+} */
```
Remove this comment.

MuhammadHasham23 replied:
Done

naveen521kk commented on src/containers/contact/Contact.css:
```diff
@@ -1,81 +1,89 @@
-    display: flex;
+  display: flex;
+}
+
+.dark-mode a {
+  color: white;
+}
+
+.dark-mode a {
+    color: white;
 }
```
Looks like duplicated.

MuhammadHasham23 replied:
Done

naveen521kk commented on src/components/header/Header.js:
```diff
@@ -1,52 +1,57 @@
+            <a href="#blogs">Blogs</a>
           </li>
           <li>
-            <a href="#achievements">Achievements</a>
+            <a href="#talks">Talks</a>
           </li>
           <li>
-            <a href="#blogs">Blogs</a>
+            <a href="#achievements">Achievements</a>
           </li>
           <li>
-            <a href="#talks">Talks</a>
+            <a href="#contact">Contact Me</a>
```
Looks like order is changed.

MuhammadHasham23 replied:
Done

naveen521kk reviewed (approved):
Some slight changes. Order that LGTM!

MuhammadHasham23 commented:
@naveen521kk all suggested fixes have been made, can we get this merged?

naveen521kk reviewed (approved):
LGTM! Waiting for either @kartikcho or @saadpasta review.

saadpasta commented:
@MuhammadHasham23 Github Cards are re-rendering please have a look. Looks like some animation issue.

naveen521kk commented on src/containers/projects/Projects.js:
```diff
@@ -1,21 +1,22 @@
   useEffect(() => {
     getRepoData();
-  }, []);
+  }, [getRepoData]);
```
Is this the issue which @saadpasta mentioned?

saadpasta commented:
Yes I guess that causing the issue not sure

On Sun, 11 Oct 2020, 3:14 pm Naveen M K, <notifications@github.com> wrote:

> *@naveen521kk* commented on this pull request.
> ------------------------------
>
> In src/containers/projects/Projects.js
> <https://github.com/saadpasta/developerFolio/pull/92#discussion_r502894970>
> :
>
> >    useEffect(() => {
>      getRepoData();
> -  }, []);
> +  }, [getRepoData]);
>
> Is this the issue which @saadpasta <https://github.com/saadpasta>
> mentioned?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/saadpasta/developerFolio/pull/92#pullrequestreview-506164459>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFR2MI5BPK2VTZVVD4NSI4TSKGARBANCNFSM4MN4SDAA>
> .
>

MuhammadHasham23 commented:
Yes, I am looking into in rn

MuhammadHasham23 commented:
It has been fixed @saadpasta 
cc: @naveen521kk

naveen521kk reviewed (approved):
LGTM!

LuisFlavioOliveira commented:
Nice job

saadpasta reviewed (approved):
LGTM. 
Thank you so much @MuhammadHasham23 for this dark mode feature your dedication was just amazing for the past 6 months.

saadpasta commented:
@MuhammadHasham23 Seeing your work on this PR and dedication to developerFolio I have invited as one the maintainer/collaborator for developerFolio. 
Welcome to the team Man ✌

naveen521kk commented:
yeah, nice work @MuhammadHasham23 it has been 6 months this PR>

@all-contributors please add @MuhammadHasham23 for code,design

naveen521kk commented:
@all-contributors please add @MuhammadHasham23 for code and design

saadpasta commented:
@naveen521kk please also add @MuhammadHasham23 in the project maintainer's tab also
